db/migrate/20210103181832_connect_ad_with_city.rb

```diff
@@ -0,0 +1,13 @@
+class ConnectAdWithCity < ActiveRecord::Migration[6.1]
+  def change
+    ActiveRecord::Base.transaction do
+      Rake::Task['db:seed_locations'].invoke
+      rename_column :ads, :city, :city_zdel
+      add_reference :ads, :city, foreign_key: true, index: true
+      Ad.all.each do |ad|
+        ad.update!(city_id: City.find_by_name(ad.city_zdel).id )
```

Comment:
This will fail if the city is not found by the name and we definitely have cities in the database on live site that don't match any city name.

Reply:
Maybe something like this would be ok

```suggestion
        city = City.find_by(name: ad.city_zdel)
        if city
          ad.update!(city_id: city.id)
        else
          ad.update!(city_id: City.first.id)
          puts "* No matching city for ad #{ad.id}."
        end
```
After @vfonic normalisation I suppose we shouldn't have lot of this and we can correct them manually.

Reply:
I spoke about this with Vlado, we need to have normalized data first in the DB, any irregularities we can manually change. idea is for this to fail so it alert us we have mismatched data that we can than alter